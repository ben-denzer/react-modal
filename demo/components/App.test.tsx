import * as React from 'react';
import { mount, shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('should render without crashing', () => {
    shallow(<App />);
  });

  it('should open modal on button click', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find('#modal-modalBg').prop('style')).toHaveProperty(
      'display',
      'none'
    );
    wrapper.find('#demo-button').simulate('click'); // open modal
    expect(wrapper.find('#modal-modalBg').prop('style')).toHaveProperty(
      'display',
      'block'
    );
  });

  it('should have the correct package name in the demo', () => {
    const wrapper = mount(<App />);
    const packageNameRegex = /@bdenzer\/react-modal/;
    const headingText = wrapper.find('h1').text();
    const bodyText = wrapper.find('#modal-modalBody').text();
    expect(packageNameRegex.test(headingText)).toBe(true);
    expect(packageNameRegex.test(bodyText)).toBe(true);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
