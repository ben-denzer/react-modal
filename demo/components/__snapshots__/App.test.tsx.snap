// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Modal
          closeModal={[Function]}
          customStyle={Object {}}
          shouldShowModal={false}
          title="Modal Demo"
        >
          <div>
            This is @bdenzer/react-modal. You can put any React Component in this body section.
          </div>
        </Modal>,
        <div
          style={
            Object {
              "fontFamily": "Arial, Helvetica, sans-serif",
              "textAlign": "center",
            }
          }
        >
          <h1>
            @bdenzer/react-modal Demo
          </h1>
          <p>
            The default styles are made to look like Bootstrap's modal
          </p>
          <button
            id="demo-button"
            onClick={[Function]}
          >
            Open Default Modal
          </button>
          <p>
            But you can easily customize just about everything about it. Here is an example of some of the things you could do.
          </p>
          <button
            id="demo-customizedModal"
            onClick={[Function]}
          >
            Open Customized Modal
          </button>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div>
              This is @bdenzer/react-modal. You can put any React Component in this body section.
            </div>,
            false,
          ],
          "closeModal": [Function],
          "customStyle": Object {},
          "shouldShowModal": false,
          "title": "Modal Demo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "This is @bdenzer/react-modal. You can put any React Component in this body section.",
            },
            "ref": null,
            "rendered": "This is @bdenzer/react-modal. You can put any React Component in this body section.",
            "type": "div",
          },
          false,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              @bdenzer/react-modal Demo
            </h1>,
            <p>
              The default styles are made to look like Bootstrap's modal
            </p>,
            <button
              id="demo-button"
              onClick={[Function]}
            >
              Open Default Modal
            </button>,
            <p>
              But you can easily customize just about everything about it. Here is an example of some of the things you could do.
            </p>,
            <button
              id="demo-customizedModal"
              onClick={[Function]}
            >
              Open Customized Modal
            </button>,
          ],
          "style": Object {
            "fontFamily": "Arial, Helvetica, sans-serif",
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "@bdenzer/react-modal Demo",
            },
            "ref": null,
            "rendered": "@bdenzer/react-modal Demo",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The default styles are made to look like Bootstrap's modal",
            },
            "ref": null,
            "rendered": "The default styles are made to look like Bootstrap's modal",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Open Default Modal",
              "id": "demo-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Open Default Modal",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "But you can easily customize just about everything about it. Here is an example of some of the things you could do.",
            },
            "ref": null,
            "rendered": "But you can easily customize just about everything about it. Here is an example of some of the things you could do.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Open Customized Modal",
              "id": "demo-customizedModal",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Open Customized Modal",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Modal
            closeModal={[Function]}
            customStyle={Object {}}
            shouldShowModal={false}
            title="Modal Demo"
          >
            <div>
              This is @bdenzer/react-modal. You can put any React Component in this body section.
            </div>
          </Modal>,
          <div
            style={
              Object {
                "fontFamily": "Arial, Helvetica, sans-serif",
                "textAlign": "center",
              }
            }
          >
            <h1>
              @bdenzer/react-modal Demo
            </h1>
            <p>
              The default styles are made to look like Bootstrap's modal
            </p>
            <button
              id="demo-button"
              onClick={[Function]}
            >
              Open Default Modal
            </button>
            <p>
              But you can easily customize just about everything about it. Here is an example of some of the things you could do.
            </p>
            <button
              id="demo-customizedModal"
              onClick={[Function]}
            >
              Open Customized Modal
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div>
                This is @bdenzer/react-modal. You can put any React Component in this body section.
              </div>,
              false,
            ],
            "closeModal": [Function],
            "customStyle": Object {},
            "shouldShowModal": false,
            "title": "Modal Demo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "This is @bdenzer/react-modal. You can put any React Component in this body section.",
              },
              "ref": null,
              "rendered": "This is @bdenzer/react-modal. You can put any React Component in this body section.",
              "type": "div",
            },
            false,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                @bdenzer/react-modal Demo
              </h1>,
              <p>
                The default styles are made to look like Bootstrap's modal
              </p>,
              <button
                id="demo-button"
                onClick={[Function]}
              >
                Open Default Modal
              </button>,
              <p>
                But you can easily customize just about everything about it. Here is an example of some of the things you could do.
              </p>,
              <button
                id="demo-customizedModal"
                onClick={[Function]}
              >
                Open Customized Modal
              </button>,
            ],
            "style": Object {
              "fontFamily": "Arial, Helvetica, sans-serif",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "@bdenzer/react-modal Demo",
              },
              "ref": null,
              "rendered": "@bdenzer/react-modal Demo",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "The default styles are made to look like Bootstrap's modal",
              },
              "ref": null,
              "rendered": "The default styles are made to look like Bootstrap's modal",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Open Default Modal",
                "id": "demo-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Open Default Modal",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "But you can easily customize just about everything about it. Here is an example of some of the things you could do.",
              },
              "ref": null,
              "rendered": "But you can easily customize just about everything about it. Here is an example of some of the things you could do.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Open Customized Modal",
                "id": "demo-customizedModal",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Open Customized Modal",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
